packages/terra-list/src/ListItem.jsx

```diff
@@ -22,57 +26,73 @@ const propTypes = {
    */
   isSelectable: PropTypes.bool,
   /**
-   * Whether or not the list item has a disclosure indicator presented.
+   * The associated metaData to be provided in the onSelect callback.
    */
-  hasChevron: PropTypes.bool,
+  // eslint-disable-next-line react/forbid-prop-types
+  metaData: PropTypes.object,
+  /**
+   * Function callback for when the appropriate click or key action is performed.
+   * Callback contains the javascript evnt and prop metadata, e.g. onSelect(event, metaData)
+   */
+  onSelect: PropTypes.func,
   /**
    * Function callback for the ref of the li.
    */
   refCallback: PropTypes.func,
 };
 
 const defaultProps = {
-  content: undefined,
+  children: [],
+  hasChevron: false,
   isSelected: false,
-  isSelectable: undefined,
-  hasChevron: undefined,
+  isSelectable: false,
 };
 
 const ListItem = ({
-  content,
+  children,
+  hasChevron,
   isSelected,
   isSelectable,
-  hasChevron,
+  metaData,
+  onBlur,
+  onClick,
+  onKeyDown,
+  onMouseDown,
+  onSelect,
   refCallback,
   ...customProps
 }) => {
   const listItemClassNames = cx([
     'list-item',
     { selected: isSelected },
     { 'is-selectable': isSelectable },
+    { 'item-has-chevron': hasChevron },
     customProps.className,
   ]);
 
-  const ariaSpread = isSelectable ? { role: 'option', 'aria-selected': isSelected } : {};
+  const attrSpread = {};
+  if (isSelectable) {
+    attrSpread.onClick = ListUtils.wrappedOnClickForItem(onClick, onSelect, metaData);
+    attrSpread.onKeyDown = ListUtils.wrappedOnKeyDownForItem(onKeyDown, onSelect, metaData);
+    attrSpread.tabIndex = '0';
+    attrSpread.role = 'option';
+    attrSpread['aria-selected'] = isSelected;
+    attrSpread['data-item-show-focus'] = 'true';
+    attrSpread.onBlur = ListUtils.wrappedEventCallback(onBlur, event => event.currentTarget.setAttribute('data-item-show-focus', 'true'));
+    attrSpread.onMouseDown = ListUtils.wrappedEventCallback(onMouseDown, event => event.currentTarget.setAttribute('data-item-show-focus', 'false'));
+  }
 
+  let childContent = children;
   if (hasChevron) {
-    const chevron = <span className={cx('chevron')}><ChevronRight height="0.8em" width="0.8em" /></span>;
-
-    return (
-      <li {...customProps} {...ariaSpread} className={listItemClassNames} ref={refCallback}>
-        <Arrange
-          fill={content}
-          fitEnd={chevron}
-          align="center"
-          fillAttributes={{ className: cx('arrange-content') }}
-        />
-      </li>
-    );
+    childContent = [
+      <div className={cx('item-fill')} key="item-fill">{childContent}</div>,
+      <div className={cx('item-end')} key="item-end">{<span className={cx('chevron')}><ChevronRight height="0.8em" width="0.8em" /></span>}</div>,
```

Comment:
I realize you're just moving this code, but do we want to be using em units here?

Reply:
@neilpfeiffer